src/sentry/search/events/datasets/base.py

```diff
@@ -55,3 +62,170 @@ def result_type_fn(
                 return argument.get_type(value)
 
         return result_type_fn
+
+
+def project_threshold_config_resolver_factory(
+    tag_value_resolver: Callable[[Optional[UseCaseKey], int, Sequence[int]], str],
+    column_name_resolver: Callable[[Optional[UseCaseKey], int, Sequence[int]], str],
+) -> Callable[[Sequence[int], int, Optional[UseCaseKey]], SelectType]:
```

Comment:
Could we just make this a helper function that accepts these two resolvers instead? I feel like a factory is overkill here when all it does is accept the callables

Reply:
I personally prefer a factory function but we could go with @wmak 's suggestion as they will be mainly maintaining this code

Reply:
Did we settle on the normal function then? @wmak

Reply:
Yes please, thanks y'all